### PR TITLE
Update Release Notes Handbook to update GitHub team

### PR DESCRIPTION
**Pull request by ramrodo**
#### What type of PR is this:

/kind documentation

#### What this PR does / why we need it:

This PR adds a note to not forget to update the release notes members at Kubernetes org repo to have a GitHub team.

#### Which issue(s) this PR fixes:

None

### CODE REVIEW

File: release-team/role-handbooks/release-notes/README.md [hunk 1]
```diff
@@ -105,6 +105,7 @@ Release Notes lead should be responsible for granting team members required acce
 1. Add OWNERS file into release notes directory. [Sample PR](https://github.com/kubernetes/sig-release/pull/1943) for v1.25 release.
 2. Add release notes team members for CHANGELOG review/approval.
    [Sample PR](https://github.com/kubernetes/kubernetes/pull/112045) for v1.25 release.
+3. Add release notes team members to the Kubernetes org as a GitHub team [Sample PR](https://github.com/kubernetes/org/pull/3735)
```
Comment (leonardpahlke) [inline release-team/role-handbooks/release-notes/README.md:108]: At the start of the cycle the release-lead updates the document - so I would rephrase this a little bit.


```suggestion
3. Typically, the release lead updates the [release notes team members](https://github.com/kubernetes/org/blob/main/config/kubernetes/sig-release/teams.yaml#L387) along with other teams in a single PR; check the PR and make sure the release notes team has been updated, otherwise open a PR yourself [sample PR](https://github.com/kubernetes/org/pull/3735)
```

**Review by leonardpahlke — CHANGES_REQUESTED** [review]

File: release-team/role-handbooks/release-notes/README.md (hunk 1 above)
Comment (reylejano) [inline release-team/role-handbooks/release-notes/README.md:108]: nit: I think a link to the file is also helpful
```suggestion
3. Add release notes team members to the `release-team-release-notes` GitHub team under `release-team` in [k/org/config/kubernetes/sig-release/teams.yaml](https://github.com/kubernetes/org/blob/main/config/kubernetes/sig-release/teams.yaml). [Sample PR](https://github.com/kubernetes/org/pull/3735)
```

**Comment by leonardpahlke** [comment]
@reylejano we reviewed at the same time, lol

File: release-team/role-handbooks/release-notes/README.md (hunk 1 above)
Comment (reylejano) [reply release-team/role-handbooks/release-notes/README.md:108]: +1 to @leonardpahlke 's suggestion to state at the start of the release.
I didn't include the Line number because it changes

**Comment by ramrodo** [comment]
@reylejano @leonardpahlke I updated the note considering both suggestions :)

**Comment by leonardpahlke** [comment]
Thank you :)
